src/components/AmountTextInput.js

```diff
@@ -27,16 +27,25 @@ class AmountTextInput extends React.Component {
       return;
     }
 
+    if (this.props.allowOnlyInteger) {
+      // We allow only integers for NFT
+      text = text.replace(/[^0-9]/g, '');
```

Comment:
I agree with the lint error, assigning to function parameter is bad

Use `let newText = (...)` to avoid side effects

Reply:
Yeah, makes sense. Done